src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleasePermissionService.cs

```diff
@@ -0,0 +1,181 @@
+                        .Where(urr =>
+                            urr.ReleaseId == release?.Id
+                            && urr.Role == Contributor)
+                        .ToListAsync();
+
+                    return allContributorsForRelease
+                        .Select(urr => new ContributorViewModel
+                        {
+                            ReleaseRoleId = urr.Id,
+                            UserId = urr.UserId,
+                            UserFullName = urr.User.DisplayName,
+                            UserEmail = urr.User.Email,
+                        })
+                        .OrderBy(model => model.UserFullName)
+                        .ToList();
+                });
+        }
+
+        public async Task<Either<ActionResult, List<ContributorViewModel>>>
+            GetPublicationContributorList(Guid releaseId)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Release>(releaseId,
+                    query =>
+                        query.Include(r => r.Publication.Releases))
+                .OnSuccessDo(release => _userService
+                    .CheckCanUpdateReleaseRole(release.Publication, Contributor))
+                .OnSuccess(async release =>
+                {
+                    var allLatestReleases = release.Publication.GetLatestVersionsOfAllReleases();
+
+                    var allLatestReleaseIds = allLatestReleases
+                        .Select(r => r.Id)
+                        .ToList();
+
+                    var allContributorReleaseRoles = await _contentDbContext.UserReleaseRoles
+                        .Include(releaseRole => releaseRole.User)
+                        .AsAsyncEnumerable()
+                        .Where(userReleaseRole =>
+                            allLatestReleaseIds.Contains(userReleaseRole.ReleaseId)
+                            && userReleaseRole.Role == Contributor)
+                        .ToListAsync();
+
+                    var allPublicationContributors = allContributorReleaseRoles
+                        .Select(releaseRole => releaseRole.User)
+                        .Distinct()
+                        .ToList();
+
+                    return allPublicationContributors
+                        .Select(user =>
+                        {
+                            var roleForThisRelease = allContributorReleaseRoles.SingleOrDefault(releaseRole =>
+                                releaseRole.UserId == user.Id
+                                && releaseRole.ReleaseId == releaseId);
+
+                            return new ContributorViewModel
+                            {
+                                UserId = user.Id,
+                                UserFullName = user.DisplayName,
+                                UserEmail = user.Email,
+                                ReleaseRoleId = roleForThisRelease?.Id,
+                            };
+                        })
```

Comment:
Discussed this with Nick in Slack. Currently, ListPublicationContributors takes a releaseId to determine whether checkboxes on are selected on the add/remove contributors page. As I understand it, this breaks the RESTful convention of the codebase by mixing concerns. 

Instead of having one endpoint that returns all contributors for a publication and whether they have contributor access for a particular release, we should use two separate endpoints - one to get a publications contributors, and another to get a release's contirbutors. The frontend is then made responsible for stitching these two together.